arm-dns/2016-04-01/swagger/dns.json

```diff
@@ -0,0 +1,1202 @@
+          "format": "int64",
+          "description": "Gets or sets the TTL of the records in the RecordSet."
+        },
+        "ARecords": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ARecord"
+          },
+          "description": "Gets or sets the list of A records in the RecordSet."
+        },
+        "AAAARecords": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AaaaRecord"
+          },
+          "description": "Gets or sets the list of AAAA records in the RecordSet."
+        },
+        "MXRecords": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MxRecord"
+          },
+          "description": "Gets or sets the list of MX records in the RecordSet."
+        },
+        "NSRecords": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NsRecord"
+          },
+          "description": "Gets or sets the list of NS records in the RecordSet."
+        },
+        "PTRRecords": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PtrRecord"
+          },
+          "description": "Gets or sets the list of PTR records in the RecordSet."
+        },
+        "SRVRecords": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SrvRecord"
+          },
+          "description": "Gets or sets the list of SRV records in the RecordSet."
+        },
+        "TXTRecords": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/TxtRecord"
+          },
+          "description": "Gets or sets the list of TXT records in the RecordSet."
+        },
+        "CNAMERecord": {
+          "$ref": "#/definitions/CnameRecord",
+          "description": "Gets or sets the CNAME record in the RecordSet."
+        },
+        "SOARecord": {
+          "$ref": "#/definitions/SoaRecord",
+          "description": "Gets or sets the SOA record in the RecordSet."
+        }
+      },
+      "description": "Represents the properties of the records in the RecordSet."
+    },
+    "RecordSet": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Gets or sets the ID of the resource."
+        },
+        "name": {
+          "type": "string",
+          "description": "Gets or sets the name of the resource."
+        },
+        "type": {
+          "type": "string",
+          "description": "Gets or sets the type of the resource."
+        },
+        "etag": {
+          "type": "string",
+          "description": "Gets or sets the ETag of the RecordSet."
+        },
+        "location": {
+          "type": "string",
+          "description": "Gets or sets the location of the resource."
+        },
+        "properties": {
+          "$ref": "#/definitions/RecordSetProperties",
+          "description": "Gets or sets the properties of the RecordSet."
```

Comment:
By doing that customers get a unified experience across all the RPs. Compute, Network, Storage, etc. are all doing that. All the breaking changes would happen one time. This will be a one time effort right now as a part of moving to AutoRest and will ease life for customers. However, I leave it up to you to decide.